### PR TITLE
Task/adq 170 medorder document model

### PR DESCRIPTION
**Pull request by jhulbert-va**
[ADQ-170](https://vasdvp.atlassian.net/browse/ADQ-170)

@vhaisdharrij08 can you check the cardinality of the values in this model

https://github.com/department-of-veterans-affairs/health-apis-data-query/pull/263

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/DatamartMedicationOrder.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  private Status status;
+
+  private Optional<Instant> dateEnded;
+
+  private DatamartReference prescriber;
+
+  private DatamartReference medication;
+
+  private List<DosageInstruction> dosageInstruction;
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/DatamartMedicationOrder.java]: Create a lazy getter that initializes to empty list if necessary.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/DatamartMedicationOrder.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  @NoArgsConstructor(access = AccessLevel.PRIVATE)
+  static final class DosageInstruction {
+
+    private Optional<String> dosageText;
+
+    private Optional<String> timingText;
+
+    private Optional<String> additionalInstructions;
+
+    private boolean asNeeded;
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/DatamartMedicationOrder.java]: optional?

File: data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/medicationorder/datamart-medication-order.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{
+  "objectType" : "MedicationOrder",
+  "objectVersion" : "1",
+  "etlDate" : "2019-07-17T18:02:04Z",
```
Comment (bschofield-va) [inline data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/medicationorder/datamart-medication-order.json]: Should not have etlDate

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/DatamartMedicationOrder.java (hunk 2 above)
Comment (jhulbert-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/DatamartMedicationOrder.java]: James came back saying `asNeeded` is always populated.

File: data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/medicationorder/datamart-medication-order-v0.json [hunk 4]
```diff
@@ -1,7 +1,6 @@
 {
   "objectType" : "MedicationOrder",
   "objectVersion" : "1",
-  "etlDate" : "2019-07-17T18:02:04Z",
```
Comment (bschofield-va) [inline data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/medicationorder/datamart-medication-order-v0.json]: This one _should_ have the etlDate. The point of the two samples is to ensure that we read both versions of the document.

**Pull request merged**